sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -712,15 +712,21 @@ static pi_result getOrCreatePlatform(ze_driver_handle_t ZeDriver,
     static auto PiPlatformsCacheMutex = new std::mutex;
 
     std::lock_guard<std::mutex> Lock(*PiPlatformsCacheMutex);
-    for (const pi_platform &CachedPlatform : *PiPlatformsCache) {
-      if (CachedPlatform->ZeDriver == ZeDriver) {
-        Platform[0] = CachedPlatform;
-        return PI_SUCCESS;
-      }
+    if (!PiPlatformsCache->empty()) {
```

Comment:
Sorry for not catching this earlier.  Here you assume that an empty cache means it hasn't been populated yet.  Just as with devices, it's probably better to assume that the system might have no platforms.  Therefore, create a new static variable `PiPlatformCachePopulated` and check that instead.  Something like:

```
static bool PiPlatformCachePopulated = false;

std::lock_guard<std::mutex> Lock(*PiPlatformsCacheMutex);
if (PiPlatformCachePopulated) {
  *PlatformCache = PiPlatformsCache;
  return PI_SUCCESS;
}
```

Reply:
Ah yes, I should thought to update this one as well. I have made this change. Thanks!